### PR TITLE
fix(skaffold): Skaffold spawns too many backends

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Companion PR

- https://github.com/Substra/hlf-k8s/pull/147
- https://github.com/Substra/orchestrator/pull/144

## Description

Update skaffold schema to `v4beta2` due to migration to Skaffold version `2.x.x` which was compatible with our schemas.

## Description

Update skaffold schema to `v4beta2` due to migration to Skaffold version `2.x.x` which was compatible with our schemas and leverages new `setValueTemplate` instead of `artifactOverrides`.

## How has this been tested?

Local tests & CI

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by ThibaultFy** [comment]
Maybe launch the e2e tests ? (do they use skafffold ?)

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by SdgJlbl** [comment]
I can confirm that this fix the bug on my local machine (with skaffold 2.1.0). 

Do we want to update the local deployment doc with this version number as the recommended one? (I can do it since I have a PR in review)

**Comment by oleobal** [comment]
> This PR updates the schema used in our Skaffold version and leverages new setValueTemplate instead of `

Instead of `artifactOverrides`?

> (do they use skafffold ?)

`run-ci` does use Skaffold for deployment

**Comment by guilhem-barthes** [comment]
Relaunched Owlfred, as it stopped at downloading Helm

> Instead of `artifactOverrides`?

Yes, [no more artifactOverrides](https://docs.substra.org/en/stable/operations/howto.html)

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: skaffold.yaml [hunk 1]
```diff
@@ -1,63 +1,71 @@
...
-            explicitRegistry: true
-        artifactOverrides:
-          server.image: substra/substra-backend
-          scheduler.image: substra/substra-backend
-          worker.image: substra/substra-backend
-          schedulerWorker.image: substra/substra-backend
-          worker.events.image: substra/substra-backend
-          api.events.image: substra/substra-backend
-
+        skipBuildDependencies: true
```
Comment (oleobal) [inline skaffold.yaml:41]: So why do we use `skipBuildDependencies` and instead have a hook for that?

Comment (guilhem-barthes) [reply skaffold.yaml:56]: Because it calls `helm dep update` and not `helm dep build`

File: skaffold.yaml [hunk 2]
```diff
@@ -1,63 +1,58 @@
...
-  statusCheckDeadlineSeconds: 600
   helm:
     releases:
       - name: backend-org-1
         chartPath: charts/substra-backend
+        valuesFiles:
+        - examples/values/backend-org-1.yaml
         namespace: org-1
+        setValues:
+          _: _
```
Comment (oleobal) [inline skaffold.yaml:29]: It might be good to preserve the comment from earlier (`# to support profiles`) else this is kinda mysterious.

**Review by oleobal — APPROVED** [review]

**Pull request merged**